consul_kv/api.py

```diff
@@ -76,33 +90,36 @@ def put_kv_txn(mapping, endpoint=DEFAULT_TXN_ENDPOINT, timeout=socket._GLOBAL_DE
         ))
 
 
-def get_kv(k=None, recurse=False, endpoint=DEFAULT_KV_ENDPOINT, timeout=socket._GLOBAL_DEFAULT_TIMEOUT):
-    """
-    Get the key value mapping from the distributed key value store
-    :param str k: key to get
-    :param bool recurse: whether or not to recurse over the path and
-    retrieve all nested values
-    :param str endpoint: API url to get the value from
-    :param int timeout: Seconds before timing out
-    :return dict mapping: key value mapping
-    """
-    url = join(endpoint, k) if k else endpoint
-    req = request.Request(
-        url=join(url, '?recurse') if recurse else url,
-        method='GET'
-    )
-    with request.urlopen(req, timeout=timeout) as r:
-        result = loads(r.read().decode('utf-8'))
-    mapping = {
-        # values are stored base64 encoded in consul, they
-        # are decoded before returned by this function.
-        r['Key']: b64decode(r['Value']).decode('utf-8')
-        if r['Value'] else None for r in result if r['Key']
-    }
-    return mapping
-
-
-def delete_kv(k=None, recurse=False, endpoint=DEFAULT_KV_ENDPOINT, timeout=socket._GLOBAL_DEFAULT_TIMEOUT):
+def get_kv_builder(postprocessor=None):
+    def get_kv_raw(k=None, recurse=False, endpoint=DEFAULT_KV_ENDPOINT, timeout=socket._GLOBAL_DEFAULT_TIMEOUT):
+        url = join(endpoint, k) if k else endpoint
+        url = "%s?recurse"  % url if recurse else url
```

Comment:
I would prefer it if you used the new style string formatting with `{}` and `.format()`